lib/widgets/payment_request_dialog.dart

```diff
@@ -195,231 +77,96 @@ class PaymentRequestDialogState extends State<PaymentRequestDialog>
-            rect: transitionAnimation,
-            child: Container(
-              height: widget.invoice.payeeImageURL.isEmpty
-                  ? _dialogContentHeight - _dialogTitleHeight : _dialogContentHeight + _dialogTitleHeight,
-              width: MediaQuery.of(context).size.width,
-              decoration: ShapeDecoration(
-                color: colorAnimation.value,
-                shape: RoundedRectangleBorder(
-                    borderRadius: BorderRadius.circular(borderAnimation.value)),
-              ),
-              child: Column(
-                  mainAxisAlignment: MainAxisAlignment.spaceAround,
-                  mainAxisSize: MainAxisSize.max,
-                  children: <Widget>[
-                    Opacity(
-                        opacity: opacityAnimation.value,
-                        child: Padding(
-                            padding: _titlePadding,
-                            child: _buildPaymentRequestTitle())),
-                    Opacity(
-                        opacity: opacityAnimation.value,
-                        child: Padding(
-                            padding: _contentPadding,
-                            child: _buildPaymentRequestContent())),
-                    Opacity(
-                        opacity: opacityAnimation.value,
-                        child: _buildPaymentRequestActions())
-                  ]),
-            ),
-          ),
-        ]));
-
-  }
-
-  Widget _buildPaymentRequestActions() {
-    if (_state == PaymentRequestState.WAITING_FOR_CONFIRMATION) {
-      List<Widget> children = <Widget>[
-        new FlatButton(
-          child: new Text("NO", style: theme.buttonStyle),
-          onPressed: () {
-            Navigator.of(context).pop();
-          },
-        ),
-        new FlatButton(
-          child: new Text("YES", style: theme.buttonStyle),
-          onPressed: () {
-            widget.accountBloc.sentPaymentsSink
-                .add(PayRequest(widget.invoice.rawPayReq, _amountToPay));
-            setState(() {
-              _state = PaymentRequestState.PROCESSING_PAYMENT;
-            });
-          },
-        ),
-      ];
-      return Padding(
-        padding: const EdgeInsets.only(bottom: 16.0,right: 8.0),
-        child: Row(
-          key: _actionsKey,
-          mainAxisSize: MainAxisSize.max,
-          mainAxisAlignment: MainAxisAlignment.end,
-          children: children,
-        ),
-      );
+    if (_state == PaymentRequestState.PROCESSING_PAYMENT) {
+      return ProcessingPaymentDialog(widget.context, widget.accountBloc, widget.firstPaymentItemKey, widget.scrollController, _initialDialogSize);
+    } else if (_state == PaymentRequestState.WAITING_FOR_CONFIRMATION) {
+      return PaymentConfirmationDialog(widget.accountBloc, widget.invoice, _initialDialogSize, _amountToPay, _amountToPayStr, (state) => _onStateChange(state));
     } else {
-      return null;
+      return _buildPaymentRequestDialog();
```

Comment:
The two other dialogs were well extracted out to their own widgets which contributes a lot to the code quality of this PR.
Any reason why the third dialog is not in its own widget?

Reply:
I believe I was confused because of the class naming, PaymentRequestDialog. I extracted this dialog into a widget called PaymentRequestInfoDialog.